src/android/com/microsoftopentechnologies/intellij/wizards/createvm/EndpointStep.java

```diff
@@ -0,0 +1,396 @@
+import javax.swing.table.TableCellRenderer;
+import java.awt.*;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.util.Vector;
+
+public class EndpointStep extends WizardStep<CreateVMWizardModel> {
+    private final VMServiceModule node;
+    private CreateVMWizardModel model;
+    private JPanel rootPanel;
+    private JList createVmStepsList;
+    private JTable endpointsTable;
+    private JComboBox portNameComboBox;
+    private JButton addButton;
+    private Project project;
+
+    public EndpointStep(final CreateVMWizardModel model, Project project, VMServiceModule node) {
+        super("Endpoint Settings", null);
+
+        this.node = node;
+        this.project = project;
+        this.model = model;
+
+        endpointsTable.setSelectionMode(ListSelectionModel.SINGLE_SELECTION);
+        model.configStepList(createVmStepsList, 4);
+
+        Endpoint[] defaultEndpoints = {
+                new Endpoint("Remote Desktop", "TCP", 3389, 3389),
+                new Endpoint("Powershell", "TCP", 5986, 5986),
+                new Endpoint("Http", "TCP", 80, 80),
+                new Endpoint("SSH", "TCP", 22, 22),
+                new Endpoint("FTP", "TCP", 21, 21),
+                new Endpoint("SMTP", "TCP", 25, 25),
+                new Endpoint("MYSQL", "TCP", 3306, 3306),
+                new Endpoint("MSSQL", "TCP", 1433, 1433),
+                new Endpoint("DNS", "TCP", 53, 53),
+                new Endpoint("POP3", "TCP", 110, 110),
+                new Endpoint("POP3S", "TCP", 995, 995),
+                new Endpoint("Https", "TCP", 443, 443),
+                new Endpoint("SMTPS", "TCP", 587, 587),
+                new Endpoint("LDAP", "TCP", 389, 389),
+                new Endpoint("IMAP", "TCP", 143, 143),
+                new Endpoint("WebDeploy", "TCP", 8172, 8172),
+        };
+
+        portNameComboBox.setModel(new DefaultComboBoxModel(defaultEndpoints));
+
+        addButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent actionEvent) {
+                EndpointTableModel endpointsTableModel = (EndpointTableModel) endpointsTable.getModel();
+
+                Endpoint endpoint = (portNameComboBox.getSelectedItem() instanceof Endpoint)
+                        ? (Endpoint) portNameComboBox.getSelectedItem()
+                        : new Endpoint(portNameComboBox.getSelectedItem().toString(), "TCP", 0, 0);
+
+                endpointsTableModel.getData().add(endpoint);
+                endpointsTableModel.fireTableDataChanged();
+            }
+        });
+
+        final EndpointTableModel endpointTableModel = new EndpointTableModel();
+        endpointTableModel.getData().add(new Endpoint("Powershell", "TCP", 5983, 5983));
+        endpointTableModel.getData().add(new Endpoint("Remote Desktop", "TCP", 3389, 3389));
```

Comment:
We should add different default endpoints depending on whether this is a Windows VM or a Linux VM.